rp2040-hal/src/timer.rs

```diff
@@ -47,6 +52,46 @@ impl Timer {
             next_end: None,
         }
     }
+
+    /// Retrieve a reference to alarm 0. Will only return a value the first time this is called
+    pub fn alarm_0(&mut self) -> Option<Alarm0> {
+        if self.alarms[0] {
```

Comment:
There could be a data race on obtaining alarms in the current setup.
Acquisition with an interrupt::free would avoid races for single core, which should be enough for now

Reply:
The `alarm_x` methods takes a mutable reference to `Timer`, does that not mean it should be safe for interrupts? The programmer will have to get a mutable reference and do locking themselves